### PR TITLE
Fix for podman when curl is up to date in the container

### PR DESCRIPTION
**Pull request by emiura**
This is a fix for a situation where openQA tries to install curl where
it is already installed (at least for cloud).

Fixes poo#90173 [qem][ec2][arm] test failure in podman

Test run at:
http://10.161.229.209/tests/221

### CODE REVIEW

File: lib/containers/utils.pm [hunk 1]
```diff
@@ -112,7 +112,7 @@ sub basic_container_tests {
     if (script_run("$runtime container exec $container_name zypper -n in curl", 300)) {
         record_info('poo#40958 - curl install failure, try with force-resolution.');
         my $output = script_output("$runtime container exec $container_name zypper in --force-resolution -y -n curl", 600);
-        die('error: curl not installed in the container') unless ($output =~ m/Installing: curl.*done/);
+        die('error: curl not installed in the container') unless (($output =~ m/Installing: curl.*done/) || ($output =~ m/The highest available.*installed/));
```
Comment (grisu48) [inline lib/containers/utils.pm:115]: Is it possible to make this specific to `curl`?

Comment (emiura) [reply lib/containers/utils.pm:115]: I can try to parse the first line "'curl' is already installed". I'm just afraid of single quotes

**Review by pdostal — APPROVED** [review]

File: lib/containers/utils.pm (hunk 1 above)
Comment (grisu48) [reply lib/containers/utils.pm:115]: You're not alone ;-)

How about something like `curl .* already installed` ?

Comment (emiura) [reply lib/containers/utils.pm]: Alright, applied your suggestion

File: lib/containers/utils.pm [hunk 2]
```diff
@@ -112,7 +112,7 @@ sub basic_container_tests {
     if (script_run("$runtime container exec $container_name zypper -n in curl", 300)) {
         record_info('poo#40958 - curl install failure, try with force-resolution.');
         my $output = script_output("$runtime container exec $container_name zypper in --force-resolution -y -n curl", 600);
-        die('error: curl not installed in the container') unless ($output =~ m/Installing: curl.*done/);
+        die('error: curl not installed in the container') unless (($output =~ m/Installing: curl.*done/) || ($output =~ m/curl .* already installed/));
```
Comment (grisu48) [inline lib/containers/utils.pm:115]:
```suggestion
        die('error: curl not installed in the container') unless (($output =~ m/Installing: curl.*done/) || ($output =~ m/\'curl\' .* already installed/));
```

I've tested it with a small test script: https://pasta.feldspaten.org/nJWr2WXw

**Review by grisu48 — CHANGES_REQUESTED** [review]

**Review by grisu48 — APPROVED** [review]
LGTM :rocket:

**Pull request merged**